test/e2e/storage/testsuites/multivolume.go

```diff
@@ -711,15 +699,26 @@ func TestConcurrentAccessToRelatedVolumes(f *framework.Framework, cs clientset.I
 		e2epod.SetAffinity(&node, actualNodeName)
 	}
 
-	// Check that all pods the same content
-	for i, pod := range pods {
-		fileName := "/mnt/volume1/index.html"
-		index := i + 1
+	for i, pvc := range pvcs {
+		var commands []string
 
+		if *pvc.Spec.VolumeMode == v1.PersistentVolumeBlock {
+			fileName := "/mnt/volume1"
+			commands = e2evolume.GenerateReadBlockCmd(fileName, len(expectedContent))
+		} else {
+			fileName := "/mnt/volume1/index.html"
+			commands = e2evolume.GenerateReadFileCmd(fileName)
+		}
+		// Check that all pods have the same content
+		index := i + 1
 		ginkgo.By(fmt.Sprintf("Checking if the volume in pod%d has expected initial content", index))
-		commands := e2evolume.GenerateReadFileCmd(fileName)
-		_, err := framework.LookForStringInPodExec(pod.Namespace, pod.Name, commands, expectedContent, time.Minute)
-		framework.ExpectNoError(err, "failed: finding the contents of the mounted file %s.", fileName)
+		_, err := framework.LookForStringInPodExec(pods[i].Namespace, pods[i].Name, commands, expectedContent, time.Minute)
```

Comment:
`i` iterates over the pvcs but it's used as an index for the pods, is this ok?

Reply:
I think so, because in the previous step we generate the PVC using that same order 🤔 

```
for i := range pvcs {
...
    podConfig := e2epod.Config{
	    NS:            ns,
	    PVCs:          []*v1.PersistentVolumeClaim{pvcs[i]},
	    SeLinuxLabel:  e2epod.GetLinuxLabel(),
	    NodeSelection: node,
	    PVCsReadOnly:  false,
	    ImageID:       e2epod.GetTestImageID(imageutils.JessieDnsutils),
    }
...
```

Or do you think it would be better idea to iterate over the pods in there too?

Reply:
So creates as many pvcs as pods, where pvcs[i] is the pvc associated with pod[i]. At least that's my reasoning 🤔 , what do you think Mau=

Reply:
sounds good, yeah there's a 1 to 1 match